app/src/main/java/com/asfoundation/wallet/home/HomeFragment.kt

```diff
@@ -1,5 +1,433 @@
+      views.headerRecyclerView.visibility = View.GONE
+      views.spacer.visibility = View.GONE
+    }
+    views.container.loadLayoutDescription(R.xml.activity_transactions_scene_short)
+  }
+
+  private fun setDefaultWalletBalance(asyncDefaultWalletBalance: Async<GlobalBalance>) {
+    when (asyncDefaultWalletBalance) {
+      Async.Uninitialized,
+      is Async.Loading -> {
+        if (asyncDefaultWalletBalance() == null) {
+          showSkeleton()
+        }
+      }
+      is Async.Fail -> {
+        onError(ErrorEnvelope(C.ErrorCode.UNKNOWN, null, asyncDefaultWalletBalance.error.throwable))
+      }
+      is Async.Success -> {
+        setWalletBalance(asyncDefaultWalletBalance())
+      }
+    }
+  }
+
+  private fun showSkeleton() {
+    views.balanceSkeleton.visibility = View.VISIBLE
+    views.balanceSkeleton.playAnimation()
+  }
+
+  private fun setWalletBalance(globalBalance: GlobalBalance) {
+    if (globalBalance.fiatValue.isNotEmpty() && globalBalance.fiatSymbol.isNotEmpty()) {
+      views.balanceSkeleton.visibility = View.GONE
+      views.balance.text = globalBalance.fiatSymbol + globalBalance.fiatValue
+      setSubtitle(globalBalance)
+    }
+  }
+
+  private fun setSubtitle(globalBalance: GlobalBalance) {
+    val subtitle = buildCurrencyString(globalBalance.appcoinsBalance, globalBalance.creditsBalance,
+        globalBalance.etherBalance, globalBalance.showAppcoins,
+        globalBalance.showCredits, globalBalance.showEthereum)
+    views.balanceSubtitle.text = Html.fromHtml(subtitle)
+  }
+
+  private fun buildCurrencyString(appcoinsBalance: Balance, creditsBalance: Balance,
+                                  ethereumBalance: Balance, showAppcoins: Boolean,
+                                  showCredits: Boolean, showEthereum: Boolean): String {
+    val stringBuilder = StringBuilder()
+    val bullet = "\u00A0\u00A0\u00A0\u2022\u00A0\u00A0\u00A0"
+    if (showCredits) {
+      val creditsString = (formatter.formatCurrency(creditsBalance.value, WalletCurrency.CREDITS)
+          + " "
+          + WalletCurrency.CREDITS.symbol)
+      stringBuilder.append(creditsString)
+          .append(bullet)
+    }
+    if (showAppcoins) {
+      val appcString = (formatter.formatCurrency(appcoinsBalance.value, WalletCurrency.APPCOINS)
+          + " "
+          + WalletCurrency.APPCOINS.symbol)
+      stringBuilder.append(appcString)
+          .append(bullet)
+    }
+    if (showEthereum) {
+      val ethString = (formatter.formatCurrency(ethereumBalance.value, WalletCurrency.ETHEREUM)
+          + " "
+          + WalletCurrency.ETHEREUM.symbol)
+      stringBuilder.append(ethString)
+          .append(bullet)
+    }
+    var subtitle = stringBuilder.toString()
+    if (stringBuilder.length > bullet.length) {
+      subtitle = stringBuilder.substring(0, stringBuilder.length - bullet.length)
+    }
+    return subtitle.replace(bullet, "<font color='#ffffff'>$bullet</font>")
```

Comment:
Since that string `bullet` is being used for verification, why would we rewrite the string again if we can use the variable and have the same result? Shouldn't we avoid duplicated code, in this case, the same string twice?